packages/cfd/src/Containers/mt5-trade-modal.tsx

```diff
@@ -1,100 +1,89 @@
 import React from 'react';
 import { DesktopWrapper, Div100vhContainer, Modal, MobileWrapper, PageOverlay, UILoader } from '@deriv/components';
-import { connect } from '../Stores/connect';
-import RootStore from '../Stores/index';
 import { localize } from '@deriv/translations';
-import { DetailsOfEachMT5Loginid } from '@deriv/api-types';
-import { TCFDDashboardContainer } from '../Components/props.types';
 import DMT5TradeModal from './dmt5-trade-modal';
 import DerivXTradeModal from './derivx-trade-modal';
+import { observer, useStore } from '@deriv/stores';
+import { TCFDPasswordReset } from './props.types';
+import { useCfdStore } from 'Stores/Modules/CFD/Helpers/useCfdStores';
 
 type TMT5TradeModalProps = {
-    mt5_trade_account: Required<DetailsOfEachMT5Loginid>;
     is_eu_user: boolean;
     is_open: boolean;
     onPasswordManager: (
         arg1: string | undefined,
         arg2: string,
-        arg3: string,
+        group: TCFDPasswordReset['account_group'],
         arg4: string,
         arg5: string | undefined
     ) => void;
     toggleModal: () => void;
-    platform: 'mt5' | 'dxtrade';
-    dxtrade_tokens: TCFDDashboardContainer['dxtrade_tokens'];
     is_demo: string;
-    show_eu_related_content: boolean;
 };
 
-const MT5TradeModal = ({
-    mt5_trade_account,
-    is_eu_user,
-    is_open,
-    onPasswordManager,
-    toggleModal,
-    dxtrade_tokens,
-    platform,
-    is_demo,
-    show_eu_related_content,
-}: TMT5TradeModalProps) => {
-    const CFDTradeModal = () => {
-        if (platform === 'mt5') {
+const MT5TradeModal = observer(
+    ({ is_eu_user, is_open, onPasswordManager, toggleModal, is_demo }: TMT5TradeModalProps) => {
+        const {
+            traders_hub: { show_eu_related_content },
+            common: { platform },
+        } = useStore();
+
+        const { dxtrade_tokens, mt5_trade_account } = useCfdStore();
```

Comment:
```suggestion
        const { traders_hub, common } = useStore();
        const { show_eu_related_content } = traders_hub;
        const { platform } = common;
        const { dxtrade_tokens, mt5_trade_account } = useCfdStore();
```
@amir-deriv I suggest object restructuring for better readability 🙇🏻 Same for other components 🙇🏻

Reply:
Done 🙏